File: Avalon/Avalon.Text/FormatArg.cs
namespace Avalon.Text;

public class FormatArg : Any
{
    public virtual long Pos { get; set; }
    public virtual long Kind { get; set; }
    public virtual Value Value { get; set; }
    public virtual bool AlignLeft { get; set; }
    public virtual long FieldWidth { get; set; }
    public virtual long MaxWidth { get; set; }
    public virtual long Base { get; set; }
    public virtual long Case { get; set; }
    public virtual long Sign { get; set; }
    public virtual uint FillChar { get; set; }
    public virtual bool HasCount { get; set; }
    public virtual long ValueCount { get; set; }
    public virtual long Count { get; set; }
}

File: Infra/Infra/FormatArg.c
#include "FormatArg.h"

InfraClassNew(FormatArg)

Int FormatArg_Init(Int o)
{
    return true;
}

Int FormatArg_Final(Int o)
{
    return true;
}

Field(FormatArg, Pos)
Field(FormatArg, Kind)
Field(FormatArg, Value)
Field(FormatArg, AlignLeft)
Field(FormatArg, FieldWidth)
Field(FormatArg, MaxWidth)
Field(FormatArg, Base)
Field(FormatArg, FillChar)
Field(FormatArg, ValueCount)
Field(FormatArg, Count)
Field(FormatArg, Form)